### PR TITLE
Add configuration for PMD, FindBugs, and Lint

### PR DESCRIPTION
**Pull request by f2rf2r**

### CODE REVIEW

**Comment by deanshi** [comment]
I can't speak to the rest of the code, but the PMD/FindBugs filters look fine by me. Although I lean more towards the "Show all possible warnings" rather than filtering out any potential useless ones.

File: src/main/groovy/io/intrepid/analysis/StaticAnalysis.groovy
Comment (gdaniels) [inline src/main/groovy/io/intrepid/analysis/StaticAnalysis.groovy]: nit: should be `createFindBugsTasks`

File: src/main/resources/default-lintConfig.xml
Comment (gdaniels) [inline src/main/resources/default-lintConfig.xml:1]: We should sync this up with https://github.com/IntrepidPursuits/sherpa/pull/78

File: README.md
Comment (gdaniels) [inline README.md]: Do you mean `lint${buildVariant}`?  Or that Android Lint will run automatically when you run `findBugs${buildVariant}`?

Comment (gdaniels) [reply README.md]: If this changes the default lint behavior, we should be crystal clear about that and exactly how it does so.

**Review by gdaniels — COMMENTED** [review]
Looks pretty much good to me, with a couple of questions (the main one being how this interacts with standard lint)

File: README.md
Comment (f2rf2r) [reply README.md]: yeah it's a typo, should be lint${buildVariant}

File: src/main/resources/default-lintConfig.xml
Comment (f2rf2r) [reply src/main/resources/default-lintConfig.xml:1]: This is up to date with the current version of config file in that PR. It's gonna be a while before @mpeng3 have time to respond to those comments and make changes. Should I wait for that before merging this PR or merge this PR now and create then v1.0.1 with the lint changes later?

Comment (gdaniels) [reply src/main/resources/default-lintConfig.xml:1]: I'm fine either way - if you agree with the suggestions over there feel free to amend this config with those updates before you merge.  In either case we should get a few projects to try it out and discuss particulars in the next few weeks.

Comment (f2rf2r) [reply src/main/resources/default-lintConfig.xml:1]: I'm gonna keep it the way it is right now since most of suggestions seems pretty minor. We can make another pr/update after we try out the current config and have a better idea of what to change

**Review by gdaniels — APPROVED** [review]
👍 on the README changes.  Will leave it to you re: lint config.

**Pull request merged**
